stdlib/public/Darwin/Accelerate/Quadrature.swift

```diff
@@ -0,0 +1,258 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Accelerate
+
+// MARK: Quadrature
+
+@available(iOS 9999, macOS 9999, tvOS 9999, watchOS 9999, *)
+/// A class that approximates the definite integral of a function over a finite interval.
+///
+/// The following code is an example of using a `Quadrature` instance to calculate the
+/// area under a curve defined by `y = sqrt(radius * radius - pow(x - radius, 2))`:
+///
+///
+///     let quadrature = Quadrature(integrator: .qags(maxIntervals: 10),
+///                                 absoluteTolerance: 1.0e-8,
+///                                 relativeTolerance: 1.0e-2)
+///
+///     let result = quadrature.integrate(over: 0.0 ... 25.0) { x in
+///         let radius: Double = 12.5
+///         return sqrt(radius * radius - pow(x - radius, 2))
+///     }
+///
+///     switch result {
+///     case .success(let integralResult, let estimatedAbsoluteError):
+///         print("quadrature success:", integralResult,
+///                                      estimatedAbsoluteError)
+///     case .failure(let error):
+///         print("quadrature error:", error.errorDescription)
+///     }
+///
+public class Quadrature {
+    
+    private var integrateOptions = quadrature_integrate_options()
+    private var integrand: ((Double) -> Double)!
+    
+    
+    /// Initializes and returns a quadrature instance.
+    ///
+    /// - Parameter integrator: An enumeration specifying the integration algorithm and relevant properties.
+    /// - Parameter absoluteTolerance: Requested absolute tolerance on the result.
+    /// - Parameter relativeTolerance: Requested relative tolerance on the result.
+    public init(integrator: Integrator,
+                absoluteTolerance: Double = 1.0e-8,
+                relativeTolerance: Double = 1.0e-2){
+        
+        self.integrator = integrator
+        integrateOptions.abs_tolerance = absoluteTolerance
+        integrateOptions.rel_tolerance = relativeTolerance
+    }
+    
+    /// The quadrature instance's integration algorithm and relevant properties.
+    public var integrator: Integrator {
```

Comment:
Should this actually be exposed as a `var`? I'm not sure that anyone will want to *change* the integrator or query it's value, so maybe the `init` is all we need, especially since creating a Quadrature object is very lightweight compared to actually doing the integration.

Reply:
Agreed. I've made that private and immutable (pending push)

Reply:
Fixed & Pushed.